google/oauth2/service_account.py

```diff
@@ -436,6 +455,26 @@ def refresh(self, request):
                 "domain wide delegation is not supported for non-default universe domain"
             )
 
+        # If a refresh is triggered under the case of credential not expired
+        # that means some abnormal cases proactively trigger this refresh, we
+        # will attemp to fetch trust boundary but tolerant on errors if we
+        # already have cached trust boundaries.
+        if (
+            self.valid
+            and self.universe_domain in _client.TRUST_BOUNDARY_ENABLED_UNIVERSES
+            and self._trust_boundary_enabled
+        ):
+            try:
+                self._trust_boundary = self.lookup_trust_boundary(request)
+            except Exception as err:
+                if not self._trust_boundary:
+                    raise err
```

Comment:
can the exception message explain that trust boundary lookup failed and a cache was not available?